### PR TITLE
mass-email

### PR DESCRIPTION
**Pull request by hiixdayah**
## Description:
- Send email to all students when announcement is created
## Checklist:
- [x] Merged latest develop
- [x] PR title makes sense

### CODE REVIEW

File: src/controllers/social.controller.ts [hunk 1]
```diff
@@ -446,4 +446,66 @@ export class SocialController {
...
+    } catch (e) {
+      logger.error('[socialController.blockUser]:' + e.message);
+      if (e.message === ERRORS.USER_DOES_NOT_EXIST) {
+        return apiResponse.error(res, httpStatusCodes.BAD_REQUEST, {
+          message: e.message,
+        });
+      } else {
+        return apiResponse.error(res, httpStatusCodes.INTERNAL_SERVER_ERROR, {
+          message: RESPONSE_ERROR.RES_ERROR,
+        });
```
Comment (kengthong) [inline src/controllers/social.controller.ts]: Can try using the new method from Utility

File: src/controllers/user.controller.ts [hunk 2]
```diff
@@ -44,6 +44,7 @@ export class UserController {
         {
           message: 'success',
           userProfile,
```
Comment (kengthong) [inline src/controllers/user.controller.ts:46]: try to standardize the return object key name across the different get user functions.
eg, user vs users vs userProfile vs newUser vs updatedUser. Easier for frontend too. I suggest using just user/newUser/updatedUser

File: src/services/user.service.ts [hunk 3]
```diff
@@ -27,23 +27,60 @@ export default class UserService {
...
+          followerId: userId,
+          followingId: accountId,
+          followingStatus: FOLLOWING_ENUM.BLOCKED,
+        },
+      });
+      //return user who blocked
+      if (followership) {
+        const userProfile = await User.findByPk(accountId, {
+          attributes: ['accountId'],
+        });
```
Comment (kengthong) [inline src/services/user.service.ts:56]: why are we returning only accountId for ah?

File: src/services/user.service.ts [hunk 4]
```diff
@@ -27,23 +27,60 @@ export default class UserService {
...
+          followerId: accountId,
+          followingId: userId,
+          followingStatus: FOLLOWING_ENUM.BLOCKED,
+        },
+      });
+      //return user who blocked
+      if (followership) {
+        const userProfile = await User.findByPk(userId, {
+          attributes: ['accountId'],
+        });
```
Comment (kengthong) [inline src/services/user.service.ts:74]: this userProfile const is declared in line 32 too. Does declaring it for a second time throw errors? Did u test the endpt?

File: src/services/user.service.ts [hunk 5]
```diff
@@ -53,7 +90,7 @@ export default class UserService {
         },
       });
       if (!followership) {
-        const privateUser = await User.findByPk(accountId, {
+        const userProfile = await User.findByPk(accountId, {
```
Comment (kengthong) [inline src/services/user.service.ts:93]: same as prev

**Review by kengthong — CHANGES_REQUESTED** [review]

File: src/services/user.service.ts (hunk 3 above)
Comment (hiixdayah) [reply src/services/user.service.ts:56]: Lawrence said for now return accountId of the user who blocked

File: src/services/user.service.ts (hunk 5 above)
Comment (hiixdayah) [reply src/services/user.service.ts:93]: yup endpoint checked, no errors

**Review by kengthong — DISMISSED** [review]

**Review by kengthong — APPROVED** [review]

**Pull request merged**